src/v/reflection/adl.h

```diff
@@ -89,10 +91,10 @@ struct adl {
                 ret.push_back(adl<value_type>{}.from(in));
             }
             return ret;
-        } else if constexpr (is_fragmented_vector) {
+        } else if constexpr (is_fragmented_vector || is_chunked_fifo) {
             using value_type = typename type::value_type;
             int32_t n = in.template consume_type<int32_t>();
-            fragmented_vector<value_type> ret;
+            type ret;
```

Comment:
nit: I guess these commits should go after https://github.com/redpanda-data/redpanda/pull/10659/commits/870528fcdbd570fbfbdbedfe738b02af8b94276f which adds the serialization in the first place